src/api/app/controllers/webui/request_controller.rb

```diff
@@ -190,14 +190,9 @@ def delete_request_dialog
   def delete_request
     req = nil
     begin
-      BsRequest.transaction do
-        opts = { target_project: params[:project], description: params[:description] }
-        opts[:target_package] = params[:package] if params[:package]
-        opts[:target_repository] = params[:repository] if params[:repository]
-        req = BsRequest.build_from_params(:delete, opts)
-        req.save!
-      end
-
+      req = BsRequest.create!(
+        description: params[:description], bs_request_actions: [BsRequestAction.new(request_action_attributes(:delete))]
```

Comment:
I would put this in a variable outside the create: `BsRequestAction.new(request_action_attributes(:delete)` in aim of readability :wink: